DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ProfileViewModel.swift
+//  DigitalLibrary
+//
+//  Created by Ognyan Yonchev on 16.02.24.
+//
+
+import SwiftUI
+
+final class ProfileViewModel: ObservableObject {
+    //@ObservedObject private var appRootManager: AppRootManager
+    let authenticationProvider: AuthenticationProvidable
+    let usersProvider: UserProvidable
+
+    @Published var user: UserModel?
+    @Published var editingField: ProfileField? = nil
+
+    @Published var email: String = ""
+    @Published var firstName: String = ""
+    @Published var lastName: String = ""
+    @Published var phone: String = ""
+
+    init(//appRootManager: AppRootManager,
+         authenticationProvider: AuthenticationProvidable,
+         usersProvider: UserProvidable) {
+        //self.appRootManager = appRootManager
+        self.authenticationProvider = authenticationProvider
+        self.usersProvider = usersProvider
+    }
+
+    func fetchCurrentUser() {
+        Task {
+            do {
+                guard let fetchedUser = await usersProvider.getCurrentUser() else {
+                    // Handle the case where no user is fetched
+                    print("No user data was fetched.")
+                    return
+                }
+                
+                DispatchQueue.main.async { [weak self] in
+                    self?.user = fetchedUser
+                    self?.email = fetchedUser.email
+                    self?.firstName = fetchedUser.firstName
+                    self?.lastName = fetchedUser.lastName
+                    self?.phone = fetchedUser.phone
+                }
+            }
+        }
+    }
+
+    func updateUserData(field: ProfileField) {
+        guard var updatingUser = user else {
+            print("Failed to update user data")
```

Comment:
We do not print, we show alerts. Check the other classses

Reply:
ok i will change it